### PR TITLE
Onboard project endpoint

### PR DESCRIPTION
**Pull request by issaafalkattan**
Adds the onboard project endpoint, however has a small bug now in auth thats being fixed in another PR where the apikey project and the projectName can be different and the endpoint will use the apikey project

### CODE REVIEW

File: manifests.yaml [hunk 1]
```diff
@@ -76,54 +76,54 @@ spec:
...
+          - name: LIGHTSTEP_COLLECTOR_HOST
+            value: lightstep-satellite.internal-services
+          - name: LIGHTSTEP_COLLECTOR_PORT
+            value: '80'
+          - name: GOOGLE_APPLICATION_CREDENTIALS
+            value: /google-credentials/credentials.json
+          - name: COGNITE_API_BASE_URL
+            valueFrom:
+              configMapKeyRef:
+                name: heimdall-scheduler
```
Comment (frikkf) [inline manifests.yaml]: This should be master one

File: package.json [hunk 2]
```diff
@@ -36,8 +36,9 @@
     "pg": "^7.18.2",
     "pg-hstore": "^2.3.3",
     "prom-client": "^12.0.0",
-    "source-map-support": "^0.5.16",
-    "testcontainers": "^2.6.0"
+    "testcontainers": "^2.6.0",
+    "sequelize": "^5.21.5",
```
Comment (frikkf) [inline package.json]: remove sequelize

File: src/api/controllers/Project/Project.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import Project from 'api/models/Project';
```
Comment (frikkf) [inline src/api/controllers/Project/Project.ts]: I guess we can remove this file/folder completely

File: src/api/controllers/Projects/Projects.ts [hunk 4]
```diff
@@ -3,20 +3,52 @@ import { APIError } from 'api/middleware/error';
 import cdfAuthService from 'services/auth/cdf';
 import clientAuthMiddleware from 'api/middleware/auth/client';
 import { AuthenticatedContext } from 'types';
+import validate from 'api/middleware/validate';
 
-import isProjectOnboarded from '../../../services/projectService';
+import ProjectServices from '../../../services/projectService';
```
Comment (frikkf) [inline src/api/controllers/Projects/Projects.ts]: nit:
if you in services create an index.ts file and inside that one do: ```export { default } from './projectsService'; ``` 

Then you can ```import ProjectsService from 'services/projectsService'```;

File: src/api/controllers/Projects/Projects.ts [hunk 5]
```diff
@@ -3,20 +3,52 @@ import { APIError } from 'api/middleware/error';
...
+
+  router.post(
+    '/',
+    clientAuthMiddleware({ cdfAuthService }),
+    validate({ body: onboardProjectSchema }),
+    async (ctx: AuthenticatedContext) => {
+      const { span, claims } = ctx.state;
+      const { body } = ctx.request;
+      try {
+        const onBoarded = await ProjectServices.onBoardProject(
```
Comment (frikkf) [inline src/api/controllers/Projects/Projects.ts]: sorry very nit picky. But onboard is a single word right? 

in that case it woudl be onboardProject()

File: src/api/controllers/Projects/validationSchema.ts [hunk 6]
```diff
@@ -3,3 +3,7 @@ import Joi from '@hapi/joi';
 export const isProjectOnboardedSchema = Joi.object({
```
Comment (frikkf) [inline src/api/controllers/Projects/validationSchema.ts]: is this being used?

File: src/db/db.ts [hunk 7]
```diff
@@ -0,0 +1,62 @@
+import { Sequelize } from 'sequelize';
```
Comment (frikkf) [inline src/db/db.ts]: This file can also be removed

File: src/services/projectService/projectService.ts [hunk 8]
```diff
@@ -20,4 +20,26 @@ const isProjectOnboarded = async (
...
+    if (project) {
+      return true;
+    }
+    throw new Error('Failed to onboard project');
+  } catch (error) {
+    throw new error('failed-precondition', error);
+  }
+};
+
+export default { isProjectOnboarded, onBoardProject };
```
Comment (frikkf) [inline src/services/projectService/projectService.ts]: Here I think we can create a class and make it a singleton.

```
class ProjectsService {

}
... bottom of the file

export default new ProjectsService();
```

File: src/api/controllers/Projects/Projects.ts (hunk 4 above)
Comment (issaafalkattan) [reply src/api/controllers/Projects/Projects.ts]: yeah there is an index.ts but i guess the '.../../../' part is unnecessary so will remove it

File: src/api/controllers/Projects/Projects.ts (hunk 5 above)
Comment (issaafalkattan) [reply src/api/controllers/Projects/Projects.ts]: true true!

**Comment by issaafalkattan** [comment]
Will close this PR because the git history got super messy, and opening a clean one with the changes you requested, review the other one instead please 🥇 FYI @frikkf

**Pull request closed without merging**